lib/teslamate_web/live/settings_live/index.html.leex

```diff
@@ -228,6 +228,27 @@ end %>
     </div>
   </div>
 
+  <div class="columns is-mobile is-centered">
+    <div class="column">
+      <h2 class="title is-4"><%= gettext("Dates") %></h2>
+      <div class="field is-horizontal">
+        <div class="field-label is-normal">
+          <%= label f, :date_format, gettext("Date Format"), class: "label" %>
+        </div>
+        <div class="field-body">
+          <div class="field">
+            <div class="control">
+              <div class="select is-fullwidth">
+                <%= select(f, :date_format, [:European, :American]) %>
```

Comment:
Since date formats vary [between countries](https://en.wikipedia.org/wiki/Date_format_by_country), I wonder if these dropdown options should be labeled differently. For example `YYYY-MM-DD` and `MM/DD/YY`.

Reply:
It would also make sense to decouple the option value from the actual text, and to make it translatable:

```suggestion
                <%= select(f, :date_format, [{gettext("European"), :European}, {gettext("American"), :American}]) %>
```

Reply:
Yep good idea on the text translations.  For the date formats, that is where I started, however the / & - don't play well with Grafana.  So I could make them YYYYMMDD & MMDDYYYY.  I tried that and eventually settled on European/American.  Am happy to go back to YYYYMMDD & MMDDYYYY if you think that is the correct thing to do?

Reply:
Got it. I mean, it's just a small detail. We do not need to rename the internal Postgres data type. Changing only the labels on the settings page would be fine. Wdyt?

Reply:
While I am glad to see some efforts are being made to make the date more flexible, I think the date format issue is a bit more complicated. 
As a UK user the American format drives me nuts. We use miles and DD/MM/YYYY as the standard formats. Neither the "American" nor the "European" (which is really ISO) formats are really any good. 

Ideally users should be able to specify their own date format or it should pick it up from the browser locale.
There may be people who want words e.g. 13-May-2021 (DD-MMM-YYYY)

Or you could have a selection of common standard ones:
- American: MM/DD/YYYY
- British: DD/MM/YYYY
- European or ISO: YY-MM-DD

Reply:
Yes you are so right. I am also a UK user, thats why i started the change, the dates annoyed me enormously. I have to say that learning about the ins and outs of  Grafana was probably the biggest disappointment I had.  The problem is that Grafana does not support the UK date format across any of its plugins or panels.  Ideally you would want to use browser locale, however this is only supported on one of the dashboards we have.

Grafana is so inconsistent with its product that its ridiculous.  I think there will be a time when it will catch up, but I can't see that being for at least a couple of years.  The only consistent formats available are DateTime_ISO and DateTime_US.

The alternative would be to replace Grafana with a product which is more feature complete, but that is a whole heap of work.

Reply:
I quite liked the idea of the texts,

- American: MM/DD/YYYY
- European or ISO: YY-MM-DD

I decided to update the date_formats type for consistency, whch now contains "ISO" & "US".

Dashboards had to be updated as a result.

Reply:
>  Ideally you would want to use browser locale, however this is only supported on one of the dashboards we have.

The `Updates` dashboard uses the "new" table component. The only reason we haven't upgraded the other dashboards is this issue I created a year ago: https://github.com/grafana/grafana/issues/24092

I don't want to get my hopes up, but currently it looks like it will finally end up in the next beta of Grafana v8. So maybe we should wait for v8.0.0-beta2 before merging this PR.